src/mlang/optimizing_ir/partial_evaluation.ml

```diff
@@ -362,8 +352,6 @@ let rec partially_evaluate_expr (ctx : partial_ev_ctx) (p : Mir.program)
           | Mast.Add, _, (Unop (Minus, e2'), _) ->
               ( Binop (Pos.same_pos_as Mast.Sub op, new_e1, e2'),
                 undef_cast d1 d2 )
-          | Mast.Add, (Literal (Float 0.), _), (_, Float) ->
-              (Pos.unmark new_e2, d2)
```

Comment:
Shouldn't this just be factorized with the ` when !Cli.optimize_unsafe_float ->` bit just down there instead of being deleted?

Reply:
It's been added to the undef case above. I've been conservative with unsafe cases.